install/kubernetes/helm/istio/charts/pilot/templates/gateway.yaml

```diff
@@ -1,3 +1,58 @@
+apiVersion: networking.istio.io/v1alpha3
+kind: Gateway
+metadata:
+  name: istio-autogenerated-k8s-ingress
```

Comment:
shouldn't the code be updated to not override this?

Reply:
It never did AFAIK (or I didn't find any code to override)...

Reply:
```virtualService := &networking.VirtualService{
			Hosts:    []string{},
			Gateways: []string{model.IstioIngressGatewayName},
		}
```

Reply:
I am very confused about this, what is it for and why is the name "autogenerated" when it is not?

Reply:
Is this because all VirtualServices that result from translating k8s are attached to this Gateway?
If yes, I suggest to rename it to "istio-reserved-k8s-ingress"

Reply:
According to:
"2. Add option to allow user to customize which gateway will handle legacy ingress"

the name should be configurable, maybe with that being the default value.

Reply:
The name is a constant in the code. Don't think it's a priority to change it, but if you have a better name please change it.

Reply:
Ideally we want to change it as part of this part, who changes the gateway from being autogenerated to being configured. it should be a simple refactoring/ rename in goland.